cpp/include/raft/distance/detail/distance.cuh

```diff
@@ -186,12 +210,49 @@ struct DistanceImpl<raft::distance::DistanceType::CosineExpanded,
            void* workspace,
            size_t worksize,
            FinalLambda fin_op,
-           cudaStream_t stream,
            bool isRowMajor,
            InType)
   {
     raft::distance::detail::cosineAlgo1<InType, AccType, OutType, FinalLambda, Index_>(
-      m, n, k, x, y, dist, (AccType*)workspace, worksize, fin_op, stream, isRowMajor);
+      m, n, k, x, y, dist, (AccType*)workspace, worksize, fin_op, handle.get_stream(), isRowMajor);
+  }
+};
+
+template <typename InType,
+          typename AccType,
+          typename OutType,
+          typename FinalLambda,
+          typename Index_>
+struct DistanceImpl<raft::distance::DistanceType::InnerProduct,
+                    InType,
+                    AccType,
+                    OutType,
+                    FinalLambda,
+                    Index_> {
+  void run(raft::device_resources const& handle,
+           const InType* x,
+           const InType* y,
+           OutType* dist,
+           Index_ m,
+           Index_ n,
+           Index_ k,
+           void* workspace,
+           size_t worksize,
+           FinalLambda fin_op,
+           bool isRowMajor,
+           InType)
+  {
+    raft::linalg::gemm(handle,
```

Comment:
Ah yes, I figured this is why we wanted to accept the handle in the distances API instead of jsut the stream- we ultimately need the cublas handle. 

I think this could be a little cleaner if we accept the `raft::resources` directly in the `distance()` function because then this file can include `raft/core/resource/cublas_handle.hpp` and none of the other files need to include it.